### ISSUE TITLE
Bug: rule `no-extra-parens` conflict with JSDoc type assertions `/**@type {}*/`

### ISSUE THREAD

**Issue by ziloen** [eslint/eslint#16386]
### Environment

Node version: 18.10.0
npm version: 8.19.2
Local ESLint version: 8.24.0
Global ESLint version: 8.24.0
Operating System: Windows 10 Enterprise LTSC

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
    "no-extra-parens": "warn"
  }
}
```
</details>
VS Code JSDoc type assertions  need to wrapp variables in parentheses

```js
function handleClick(/**@type {MouseEvent}*/ event) {
  const span = /**@type {HTMLSpanElement}*/(event.currentTarget) 
}
```

### What did you expect to happen?

Preserve the parentheses where  JSDoc type assertions

### What actually happened?

The parentheses are removed

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @ziloen, thanks for the issue!

> ```js
> function handleClick(/**@type {MouseEvent}*/ event) {
>   const span = /**@type {HTMLSpanElement}*/(event.currentTarget) 
> }
> ```

Since these parentheses are only required for a specific tool, and are generally not needed in JS code, I think this is beyond the scope of the core `no-extra-parens` rule.

We could maybe consider adding a generic option that would allow parentheses preceded by a comment that matches a regular expression. Curious what other team members think about this.

**Comment by snitin315** [comment]
I agree that this seems like out of the scope of the `no-extra-parens` rule. 

But usage like this is common with tools like JSDoc, for example, we have many instances like this in the [webpack-dev-server](https://github.com/webpack/webpack-dev-server/blob/93d90a0addbdae4cdfac301b4e9e3e5d8af70215/lib/Server.js#L221) project.

[Playground Link](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogW1wiZXJyb3JcIl0gKi9cblxuaWYgKC8qKiBAdHlwZSB7Q29tcGlsZXIgfCBNdWx0aUNvbXBpbGVyfSAqLyhvcHRpb25zKS5ob29rcykge1xuICAgICAgdXRpbC5kZXByZWNhdGUoXG4gICAgICAgICgpID0+IHt9LFxuICAgICAgICBcIlVzaW5nICdjb21waWxlcicgYXMgdGhlIGZpcnN0IGFyZ3VtZW50IGlzIGRlcHJlY2F0ZWQuIFBsZWFzZSB1c2UgJ29wdGlvbnMnIGFzIHRoZSBmaXJzdCBhcmd1bWVudCBhbmQgJ2NvbXBpbGVyJyBhcyB0aGUgc2Vjb25kIGFyZ3VtZW50LlwiLFxuICAgICAgICBcIkRFUF9XRUJQQUNLX0RFVl9TRVJWRVJfQ09OU1RSVUNUT1JcIlxuICAgICAgKSgpO1xuXG4gICAgICBbb3B0aW9ucyA9IHt9LCBjb21waWxlcl0gPSBbY29tcGlsZXIsIG9wdGlvbnNdO1xufVxuXG52YWxpZGF0ZSgvKiogQHR5cGUge1NjaGVtYX0gKi8gKHNjaGVtYSksIG9wdGlvbnMsIHtcbiAgbmFtZTogXCJEZXYgU2VydmVyXCIsXG4gIGJhc2VEYXRhUGF0aDogXCJvcHRpb25zXCIsXG59KTtcblxudGhpcy5jb21waWxlciA9IC8qKiBAdHlwZSB7Q29tcGlsZXIgfCBNdWx0aUNvbXBpbGVyfSAqLyAoY29tcGlsZXIpO1xuXG5mdW5jdGlvbiBoYW5kbGVDbGljaygvKipAdHlwZSB7TW91c2VFdmVudH0qLyBldmVudCkge1xuICBjb25zdCBzcGFuID0gLyoqQHR5cGUge0hUTUxTcGFuRWxlbWVudH0qLyhldmVudC5jdXJyZW50VGFyZ2V0KSBcbn1cblxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

> We could maybe consider adding a generic option that would allow parentheses preceded by a comment that matches a regular expression. Curious what other team members think about this.

I would be in favour of adding a option for this 👍🏻

**Comment by snitin315** [comment]
I'll work on this.
